File: src/cmd/groups.zig
const std = @import("std");
const tool_mod = @import("../tool.zig");
const output = @import("../ui/output.zig");

const help =
    \\Usage: dot groups
    \\
    \\List all tool groups with their tool counts.
    \\
    \\Options:
    \\  --help, -h    Show this help
    \\
    \\Examples:
    \\  dot groups
    \\  dot list -g k8s        # filter list by group
    \\  dot install -g k8s     # install all tools in a group
    \\
;

const GroupDesc = struct { group: tool_mod.Group, desc: []const u8 };

const group_descs = [_]GroupDesc{
    .{ .group = .k8s, .desc = "Kubernetes ecosystem" },
    .{ .group = .cloud, .desc = "Public cloud CLIs" },
    .{ .group = .iac, .desc = "Infrastructure as Code" },
    .{ .group = .containers, .desc = "Container engines" },
    .{ .group = .cm, .desc = "Configuration management" },
    .{ .group = .security, .desc = "Security scanning & secrets" },
    .{ .group = .utils, .desc = "General-purpose CLI utilities" },
    .{ .group = .terminal, .desc = "Terminal & shell UX" },
};

pub fn run(
    allocator: std.mem.Allocator,
    args: []const []const u8,
    tools: []const tool_mod.Tool,
) !void {
    _ = allocator;
    for (args) |a| {
        if (std.mem.eql(u8, a, "--help") or std.mem.eql(u8, a, "-h")) {
            output.printRaw(help);
            return;
        }
    }

    output.printSectionHeader("Groups");
    output.printFmt("\n{s}{s:<16} {s:<7} Description{s}\n", .{ output.bold, "Group", "Tools", output.reset });

    for (group_descs) |gd| {
        var count: usize = 0;
        for (tools) |t| {
            for (t.groups) |g| {
                if (g == gd.group) { count += 1; break; }
            }
        }
        output.printFmt("{s:<16} {d:<7} {s}\n", .{ @tagName(gd.group), count, gd.desc });
    }

    output.printFmt("\n{s}Tip:{s} 'dot list -g <group>'  ·  'dot install -g <group>'\n\n", .{ output.dim, output.reset });
}

File: src/cmd/groups.rs
use crate::tool::{Group, Tool};
use crate::ui::output;
use clap::Args;
use console::style;

#[derive(Debug, Args)]
pub struct GroupsArgs {}

const GROUP_DESCS: &[(Group, &str)] = &[
    (Group::K8s, "Kubernetes ecosystem"),
    (Group::Cloud, "Public cloud CLIs"),
    (Group::Iac, "Infrastructure as Code"),
    (Group::Containers, "Container engines"),
    (Group::Cm, "Configuration management"),
    (Group::Security, "Security scanning & secrets"),
    (Group::Utils, "General-purpose CLI utilities"),
    (Group::Terminal, "Terminal & shell UX"),
];

pub fn run(_args: &GroupsArgs, tools: &[Tool]) -> anyhow::Result<()> {
    output::print_section_header("Groups");

    let colored = output::get_render_mode() == output::RenderMode::Rich;
    if colored {
        eprintln!(
            "\n{:<16} {:<7} Description",
            style("Group").bold(),
            style("Tools").bold()
        );
    } else {
        eprintln!("\n{:<16} {:<7} Description", "Group", "Tools");
    }

    for (group, desc) in GROUP_DESCS {
        let count = tools.iter().filter(|t| t.groups.contains(group)).count();
        eprintln!("{:<16} {:<7} {desc}", group.name(), count);
    }

    eprintln!("\nTip: 'dot list -g <group>'  ·  'dot install -g <group>'\n");
    Ok(())
}
